src/components/widgets/support-case-table-filter.tsx

```diff
@@ -0,0 +1,312 @@
+import { ToolbarGroup } from '@patternfly/react-core/dist/dynamic/components/Toolbar';
+import {
+  MenuToggle,
+  MenuToggleElement,
+} from '@patternfly/react-core/dist/dynamic/components/MenuToggle';
+import { Select } from '@patternfly/react-core/dist/dynamic/components/Select';
+import { SelectList } from '@patternfly/react-core/dist/dynamic/components/Select';
+import { SelectOption } from '@patternfly/react-core/dist/dynamic/components/Select';
+import FilterIcon from '@patternfly/react-icons/dist/esm/icons/filter-icon';
+import { severityTypes, statusTypes } from '../../utils/consts';
+
+export const SupportCaseWidgetTableFilter: React.FunctionComponent = () => {
+  const [isProductFamilyExpanded, setIsProductFamilyExpanded] =
+    React.useState(false);
+  const [isSeverityExpanded, setIsSeverityExpanded] = React.useState(false);
+  const [isStatusExpanded, setIsStatusExpanded] = React.useState(false);
+  const [filters, setFilters] = React.useState({
+    productFamily: [''],
+    severity: [''],
+    status: [''],
+  });
+
+  const onSelect = (
+    type: string,
+    event: React.MouseEvent<Element, MouseEvent> | React.ChangeEvent<Element>,
+    selection: string
+  ) => {
+    const checked = (event.target as HTMLInputElement).checked;
+    setFilters((prev) => {
+      const prevSelections = prev[type];
+      return {
+        ...prev,
+        [type]: checked
+          ? [...prevSelections, selection]
+          : prevSelections.filter((value: string) => value !== selection),
+      };
+    });
+  };
+
+  const onProductFamilySelect = (
+    event: React.MouseEvent<Element, MouseEvent> | React.ChangeEvent<Element>,
+    selection: string
+  ) => {
+    onSelect('productFamily', event, selection);
+  };
+
+  const onSeveritySelect = (
+    event: React.MouseEvent | React.ChangeEvent,
+    selection: string
+  ) => {
+    onSelect('severity', event, selection);
+  };
+
+  const onStatusSelect = (
+    event: React.MouseEvent | React.ChangeEvent,
+    selection: string
+  ) => {
+    onSelect('status', event, selection);
+  };
+
+  const onDelete = (type: string, id: string) => {
+    if (type === 'Product Family') {
+      setFilters({
+        productFamily: filters.productFamily.filter(
+          (fil: string) => fil !== id
+        ),
+        severity: filters.severity,
+        status: filters.status,
+      });
+    } else if (type === 'Severity') {
+      setFilters({
+        productFamily: filters.productFamily,
+        severity: filters.severity.filter((fil: string) => fil !== id),
+        status: filters.status,
+      });
+    } else if (type === 'Status') {
+      setFilters({
+        productFamily: filters.productFamily,
+        severity: filters.severity,
+        status: filters.status.filter((fil: string) => fil !== id),
+      });
+    } else {
+      setFilters({ productFamily: [], severity: [], status: [] });
+    }
+  };
```

Comment:
```suggestion
const onDelete = (type: string, id: string) => {
  const filterTypes: { [key: string]: string[] } = {
    'Product Family': filters.productFamily.filter((fil: string) => fil !== id),
    'Severity': filters.severity.filter((fil: string) => fil !== id),
    'Status': filters.status.filter((fil: string) => fil !== id),
  };

  setFilters({
    productFamily: type === 'Product Family' ? filterTypes[type] : filters.productFamily,
    severity: type === 'Severity' ? filterTypes[type] : filters.severity,
    status: type === 'Status' ? filterTypes[type] : filters.status,
  });
};
```

something like this could do the same